src/MIDebugEngine/Engine.Impl/Variables.cs

```diff
@@ -90,6 +90,23 @@ internal sealed class VariableInformation : IVariableInformation
         private string DisplayHint { get; set; }
         public bool IsPreformatted { get; set; }
 
+        public string Address()
+        {
+            // ask GDB to evaluate "&expression"
+            string command = "&("+_strippedName+")";
+            IVariableInformation execVariable = new VariableInformation(command, this);
+            execVariable.SyncEval();
+            return _engine.DebuggedProcess.Natvis.FormatDisplayString(execVariable);
+        }
+
+        public uint Size()
+        {
+            // ask GDB to evaluate "sizeof(expression)"
+            string command = "sizeof("+_strippedName+")";
+            IVariableInformation execVariable = new VariableInformation(command, this);
+            execVariable.SyncEval();
+            return Convert.ToUInt32(_engine.DebuggedProcess.Natvis.FormatDisplayString(execVariable), CultureInfo.InvariantCulture);
```

Comment:
>ToUInt32 [](start = 27, length = 8)

Does this produce the right thing if GDB is in hex mode?

Reply:
I am a bit confused as to what is meant by hex mode. Is it simply having Hexadecimal Display checked as is shown below?
![image](https://user-images.githubusercontent.com/60245970/109054804-2c463200-7693-11eb-8eb0-d7107fa234b8.png)

Or do you mean running said command with the format letter x as stated here: https://ftp.gnu.org/old-gnu/Manuals/gdb/html_node/gdb_54.html? Note that I am now using the MI command directly.

Reply:
Yes, it should be triggered by checking that checkbox and doing an evaluation. That will cause a call to [SetRadix](https://github.com/Microsoft/MIEngine/blob/0e48a887feb19cf525025fa8bd5788464b667065/src/MICore/CommandFactories/MICommandFactory.cs#L300-L300) with 16, which then changes the default output radix.